### ISSUE TITLE
Focus unexpectedly jumps to primary screen when cycling focus (intermittent)

### ISSUE THREAD

**Issue by pix0r** [ianyh/Amethyst#1466]
**Describe the bug**
I'm using a two monitor setup, and have multiple windows open on each screen. Both set to "Two Pane" layout. When I have windows from two different apps open on the second screen, and one is focused, the "Move focus counter clockwise" and "Move focus clockwise" keyboard shortcuts cause the focus to unexpectedly jump back to a window on the first screen, instead of cycling focus between the windows on the second screen. When I have multiple windows of the same application open on the second screen (e.g. Safari or Brave browser), the focus stays on the second screen as expected.

**Applications:**
This seems to affect many apps. It seems like any time there are windows from different apps on the second screen, this bug appears. 

Some combinations I've tested:

Safari + Safari => no bug
Brave + Brave => no bug
Safari + Brave => bug
Brave + Slack => bug
Brave + Notes => bug

**To Reproduce**
* Set up two monitors, both using "Two Pane" layout (this does however affect all layouts as far as I can tell).
* Open four Safari windows, two on each screen
* Use commands to move focus around between windows (opt+shift+W/E to switch screens, opt+shift+J/K to move focus on same screen) - works as expected
* Replace one Safari window on second screen, with Notes app (leaving Safari as the "main" window on the left)
* Click to manually focus Safari window on second screen
* opt+shift+J to move focus to Notes window on second screen; this moves focus as expected
* opt+shift+J to move focus back to Safari window on second screen; this moves focus back to first screen!

NOTE: This behavior begins happening randomly; when I relaunch Amethyst and try to reproduce it seems to work as expected.

**Expected behavior**
Expected focus to remain on second screen, cycling back to the Safari window

**Screenshots**

**Versions:**
 - macOS: 13.2.1
 - Amethyst: 0.19.0 (103)
 
**Debug Info**
```
$ /Applications/Amethyst.app/Contents/MacOS/Amethyst --debug-info --include-apps
Version: 0.19.0 (103)

OS version: Version 13.2.1 (Build 22D68)

Screens:
	(0.0, 0.0, 3360.0, 1890.0) [(0.0, 0.0, 3360.0, 1890.0)]
	(-1728.0, -168.0, 1728.0, 1117.0) [(-1728.0, 941.0, 1728.0, 1117.0)]

Configuration:
screen-padding-bottom: 0
mod2: (
    option,
    shift,
    control
)
layouts: (
    "two-pane",
    "middle-wide",
    column,
    tall,
    "tall-right",
    wide,
    "3column-left",
    bsp,
    fullscreen
)
screen-padding-top: 0
use-canary-build: 0
restore-layouts-on-launch: 1
focus-follows-mouse: 0
window-margins: 0
ignore-menu-bar: 0
window-max-count: 0
floating: (
        {
        id = "us.zoom.xos";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.Music";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.MobileSMS";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.calculator";
        "window-titles" =         (
        );
    },
        {
        id = "org.pqrs.Karabiner-Elements.Settings";
        "window-titles" =         (
        );
    },
        {
        id = "com.1password.1password";
        "window-titles" =         (
        );
    },
        {
        id = "org.pqrs.Karabiner-EventViewer";
        "window-titles" =         (
        );
    },
        {
        id = "com.trainerroad.mac";
        "window-titles" =         (
        );
    },
        {
        id = "com.xwavesoft.pomodoromac";
        "window-titles" =         (
        );
    },
        {
        id = "com.apple.calculator";
        "window-titles" =         (
        );
    },
        {
        id = "com.grailr.CARROT-Weather";
        "window-titles" =         (
        );
    }
)
enables-layout-hud: 1
mouse-swaps-windows: 0
mod1: (
    option,
    shift
)
new-windows-to-main: 0
mouse-resizes-windows: 0
enables-layout-hud-on-space-change: 1
window-minimum-width: 0
screen-padding-left: 10
window-margin-size: 0
window-resize-step: 5
disable-padding-on-builtin-display: 0
float-small-windows: 1
window-minimum-height: 0
debug-layout-info: 0
screen-padding-right: 10
follow-space-thrown-windows: 1
mouse-follows-focus: 0
floating-is-blacklist: 1

Manageable applications:
	Brave Browser (com.brave.Browser)
	1Password (com.1password.1password)
	iTerm2 (com.googlecode.iterm2)
	Finder (com.apple.finder)
	Preview (com.apple.Preview)
	Code (com.microsoft.VSCode)
	Music (com.apple.Music)
	Safari (com.apple.Safari)
	Slack (com.tinyspeck.slackmacgap)

```
Note: `--include-apps` will list your manageable applications, but is optional if you don't want to list that.

**Additional context**
Amethyst is crashing for me regularly when I open Zoom, so I'm frequently relaunching it. This issue was happening to me reliably before, but after I've just relaunched, I'm not currently able to reproduce.

**Comment by uygarkurt** [comment]
Can you try with terminal too? I'm having a similar problem with the terminal window.

**Comment by platobms** [comment]
I have also encountered this issue, and I am using both Brave and Slack on the second monitor. When I try to move focus through the windows (`mod1` + `j`/`k`) of my second monitor the focus unexpectedly jumps back to the primary screen. 

Second Screen               | Outcome
Brave + Brave                | jumps to primary
Brave + Brave + Slack  |  jumps to primary
finder + finder              |  no issue, cycles through both
brave + finder               | no issue, cycles through both
brave + brave + finder | jumps to primary


This bug is present regardless of window layout (column vs fullscreen).
I have "Displays have separate spaces" checked (for additional context).

**Comment by jheroy** [comment]
Same issue.

**Comment by varunaswal** [comment]
I believe this issue may be due to how macos behaves, as there are similar issues with `hs.window.focus()` in Hammerspoon. It seems to happen when focusing an application that has multiple windows - it does not always focus the correct window.

I can't find the thread right now but someone suggested a workaround of first activating the application and then focusing the correct window. Maybe this is a possible fix.

Edit: found the thread - https://github.com/Hammerspoon/hammerspoon/issues/370

**Comment by varunaswal** [comment]
This does indeed seem to be a bug in either macos or chromium, as I can replicate the same behaviour using native mac space-switching shortcuts as follows:
<pre>
Screen A
   |      \
Space 1 Space 2
   |       |
Chrome1  VSCode

Screen B
   | 
Space 3
   |
Chrome2 </pre>
 1. Start with focus on  `Chrome 2` and `Screen A` on `Space 1`. 
 2. Switch to `Space 2` with `ctrl-2`. Observe that `VSCode` is now focused
 3. Switch bace to `Space 1` with `ctrl-1`. Observe that focus jumps to `Chrome2` instead of `Chrome1`
 
This was all with Amethyst disabled. I have not tested with other applications.

**Comment by ianyh** [comment]
This should be fixed in 0.20.0.
